tests/ui_tools/test_boxes.py

```diff
@@ -217,6 +217,34 @@ def test_generic_autocomplete_emojis(self, write_box, text,
         typeahead_string = write_box.generic_autocomplete(text, state)
         assert typeahead_string == required_typeahead
 
+    @pytest.mark.parametrize(['key', 'current_typeahead_mode',
+                             'expected_typeahead_mode', 'set_footer_called'], [
```

Comment:
Doesn't `set_footer_called` represent `expect_footer_was_reset` or similar?
(partly that's since `set_footer` is called elsewhere and doesn't just reset it, so this is confusing?)